### PR TITLE
Use relative time in ChapterHolder

### PR DESCRIPTION
**Pull request by ghostbear**
Screenshot_1628970378
Screenshot_1628970388
Screenshot_1628970373

### CODE REVIEW

**Comment by arkon** [comment]
Does this still respect the user preference for date format when it's beyond the "relative" threshold?

**Comment by Soitora** [comment]
> Does this still respect the user preference for date format when it's beyond the "relative" threshold?

It does not

**Comment by Soitora** [comment]
Could extensions with a known reliability in displaying chapter upload dates be able to show even more precise data?

Like this on MangaDex: 
image

Shows like this:
image

Maybe if extensions were able to pass a "hasReliableDates" or something idk

**Comment by ghostbear** [comment]
> Does this still respect the user preference for date format when it's beyond the "relative" threshold?

I have made an extension function that gets the relative time instead of using DateUtils and it uses the user preference for date format when it's beyond the "relative" threshold

It also displays "Now", could be something else like "Soon" for example, instead of "in X hours" mentioned in https://github.com/Jays2Kings/tachiyomiJ2K/issues/958

**Comment by Soitora** [comment]
> I have made an extension function that gets the relative time instead of using DateUtils and it uses the user preference for date format when it's beyond the "relative" threshold

No hours/minutes ago?

> 
> It also displays "Now", could be something else like "Soon" for example, instead of "in X hours" mentioned in [Jays2Kings#958](https://github.com/Jays2Kings/tachiyomiJ2K/issues/958)

Maybe "Recently" for anything under 1 hour and then `x hours ago` ? Or for anything under x amount of minutes relatively?

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -186,6 +186,13 @@
     <string name="pref_label_nsfw_extension">Label in extensions list</string>
     <string name="parental_controls_info">This does not prevent unofficial or potentially incorrectly flagged extensions from surfacing NSFW (18+) content within the app.</string>
 
+    <string name="recently">Recently</string>
+    <string name="today">Today</string>
+    <string name="yesterday">Yesterday</string>
+    <string name="days_ago">%1$.0f days ago</string>
```
Comment (arkon) [inline app/src/main/res/values/strings.xml:192]: I'd recommend using a plural for this with the number being the number of days (i.e. 0 = today, 1 = yesterday, other = n days ago). I'm not entirely sure how these sort of things end up getting localized in other languages.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -186,6 +186,13 @@
...
+    <string name="relative_format">Shorted relative time</string>
```
Comment (arkon) [inline app/src/main/res/values/strings.xml:193]: "Relative timestamps"?

File: app/src/main/java/eu/kanade/tachiyomi/util/lang/DateExtensions.kt [hunk 3]
```diff
@@ -94,3 +96,23 @@ fun Long.toLocalCalendar(): Calendar? {
         )
     }
 }
+
+private const val MILLISECONDS_IN_DAY = 86_400_000.0
+
+fun Date.getRelativeTime(
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/util/lang/DateExtensions.kt:102]: `toRelativeString`?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -186,6 +186,13 @@
...
+    <string name="relative_format_summary">Excludes \"X days ago\"</string>
```
Comment (arkon) [inline app/src/main/res/values/strings.xml:194]: Maybe default it to true and have the summary be an example?

**Pull request merged**